### DISCUSSION TITLE
Reactive forms should have better animations and transitions. It's not very eye pleasing when the layout shifts instantaniously.

### DISCUSSION THREAD

**Discussion by itsemon245** [filamentphp/filament#15200] [Ideas]
I love filamentphp however there is no animation on reactive form changes or no way to add some animations (at least I didn't find any in the docs). It would be very helpful if there is a way to just quickly add some basic lifecycle transitions and animations or at least a mechanism to implement custom animations.

**Comment by leandrocfe** [comment]
I believe animations or transitions are outside the scope of features Filament should directly support. Maybe you could achieve it using a [custom theme](https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme) or using [render hooks](https://filamentphp.com/docs/3.x/support/render-hooks#overview) to inject custom HTML, Livewire components and more.
